### PR TITLE
[WIP]#2559 Routing automatic loading

### PR DESCRIPTION
**Pull request by aRn0D**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | no
| License       | MIT
| Doc PR        | -

Some nodes have been added to the configuration, they are used to configure the both router.
```
sylius_resource:
    router:
        generic:
            sylius:
                prefix: prefix
        api:
            sylius:
                prefix: prefix
```
The router need to be activated :
```
sylius_routes:
    resource: .
    type: sylius
sylius_routes_api:
    resource: .
    type: sylius.api
```

### CODE REVIEW

**Comment by aRn0D** [comment]
@Sylius/core-team / @michalmarcinkowski it is time to review it !!

**Comment by aRn0D** [comment]
ping @Sylius/core-team / @michalmarcinkowski

**Comment by steffenbrem** [comment]
:+1: Nice, I like this alot!

**Comment by aRn0D** [comment]
ping @Sylius/core-team / @michalmarcinkowski / @stloyd

**Comment by aRn0D** [comment]
ping @pjedrzejewski

File: src/Sylius/Bundle/ResourceBundle/Routing/Loader/GenericLoader.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function createResourceRoutes($resource)
+    {
+        $this->collectionBuilder->add($resource, 'index', array('GET'));
+        $this->collectionBuilder->add($resource, 'show', array('GET'));
+        $this->collectionBuilder->add($resource, 'create', array('POST', 'GET'));
+        $this->collectionBuilder->add($resource, 'update', array('POST', 'GET'));
```
Comment (Strontium-90) [inline src/Sylius/Bundle/ResourceBundle/Routing/Loader/GenericLoader.php]: PUT?

**Comment by aRn0D** [comment]
ping @pjedrzejewski

**Comment by aRn0D** [comment]
ping @pjedrzejewski

**Comment by aRn0D** [comment]
ping @pjedrzejewski

**Comment by aRn0D** [comment]
ping @pjedrzejewski

**Comment by pjedrzejewski** [comment]
I have something like that in works (on grids) branch, but configured in routing, which is the right place for that I think. I think putting routing in semantic bundle config is a bit too much.

**Pull request closed without merging**